Src/Particle/AMReX_ParticleTransformation.H

```diff
@@ -394,12 +394,55 @@ auto filterParticles (DstTile& dst, const SrcTile& src, Pred&& p,
     amrex::ParallelForRNG(np,
     [=] AMREX_GPU_DEVICE (int i, amrex::RandomEngine const& engine) noexcept
     {
-        p_mask[i] = particle_detail::call_f(p, src_data, i, engine);
+        p_mask[i] = particle_detail::call_f(p, src_data, src_start+i, engine);
     });
-
+    Gpu::streamSynchronize();
```

Comment:
In versions of these that are implemented in terms of another overload, I'm not sure these syncs are needed, because the inner-most function that actually implements the copy will do so.

Reply:
Pushed update commit that removes synchronizes in outer function